infra/core/search/search-private-link.bicep

```diff
@@ -0,0 +1,24 @@
+param name string
+param searchName string
+param resourceId string
```

Comment:
probably longer name here to signify which resource.

Reply:
You can use a variety of resources in theory, see https://learn.microsoft.com/en-us/azure/search/search-indexer-howto-access-private?tabs=portal-create#supported-resource-types
I'll link to that and add a description